examples/src/main/scala/examples/hybrid/history/HybridHistory.scala

```diff
@@ -450,17 +465,24 @@ class HybridHistory(val storage: HistoryStorage,
     chainBack(storage.bestPosBlock, isGenesis).get.map(_._2).map(Base58.encode).mkString(",")
   }
 
-  //todo: real impl
   override def reportSemanticValidity(modifier: HybridBlock,
                                       valid: Boolean,
                                       lastApplied: ModifierId): (HybridHistory, ProgressInfo[HybridBlock]) = {
-    this -> ProgressInfo(None, Seq(), None, Seq())
-  }
+    val v = if (valid) Valid else Invalid
+    storage.updateValidity(modifier, v)
 
-  //todo: real impl
-  override def isSemanticallyValid(modifierId: ModifierId): ModifierSemanticValidity = {
```

Comment:
... or change the type of `v` to `Boolean` in `storage.updateValidity`.

Just so that we don't have to convert from `Boolean` to `Valid` or `Invalid`.